File: lottery-ui/src/api/train.js
/**
 * 模型训练 API
 * 提供训练任务提交和状态查询
 */
import request from '@/utils/request'

/**
 * 提交训练任务
 * @param {Object} data - 训练参数
 * @param {string} data.lottery_type - 'ssq' 或 'dlt'
 * @param {number} data.epochs - 训练轮数
 * @param {number} data.batch_size - 批次大小
 * @param {number} data.seq_len - 序列长度
 * @param {number} data.learning_rate - 学习率
 * @returns {Promise} 返回 { task_id, status, message }
 */
export function startTraining(data) {
  return request({
    url: '/train/start',
    method: 'post',
    data
  })
}

/**
 * 查询训练任务状态
 * @param {string} taskId - 任务 ID
 * @returns {Promise} 返回 { status, result }
 */
export function getTrainStatus(taskId) {
  return request({
    url: `/train/status/${taskId}`,
    method: 'get'
  })
}

File: app/api/routes/train.py
# -*- coding: utf-8 -*-
"""
模型训练接口：提交训练任务、查询训练状态。
"""
from fastapi import APIRouter, Depends, HTTPException
from celery.result import AsyncResult
from app.schemas.lottery import TrainRequest, TrainResponse
from app.tasks.training_jobs import start_training
from app.core.celery_app import celery_app

router = APIRouter(prefix="/train", tags=["Training"])

@router.post("/start", response_model=TrainResponse)
async def start_train(request: TrainRequest):
    """
    启动异步训练任务。
    提交后立即返回 task_id，用于后续查询进度。
    """
    task = start_training.delay(
        lottery_type=request.lottery_type,
        epochs=request.epochs,
        batch_size=request.batch_size,
        seq_len=request.seq_len,
        learning_rate=request.learning_rate
    )
    return TrainResponse(
        task_id=task.id,
        status="pending",
        message="训练任务已提交，请使用 task_id 查询状态"
    )

@router.get("/status/{task_id}")
async def get_train_status(task_id: str):
    """
    查询训练任务状态。
    返回状态信息（pending/running/completed/failed）及结果。
    """
    task = AsyncResult(task_id, app=celery_app)
    if task.state == 'PENDING':
        return {"task_id": task_id, "status": "pending", "result": None}
    elif task.state == 'RUNNING':
        return {"task_id": task_id, "status": "running", "result": task.info}
    elif task.state == 'SUCCESS':
        return {"task_id": task_id, "status": "completed", "result": task.result}
    elif task.state == 'FAILURE':
        return {"task_id": task_id, "status": "failed", "result": str(task.info)}
    else:
        return {"task_id": task_id, "status": task.state, "result": None}